File: src/Samvaad/services/polyglot/hello.pl
#!/usr/bin/perl
use strict;
use warnings;

my $input = $ARGV[0] || '';

if (!$input) {
    print '{"error": "No arguments provided"}';
    exit;
}

my $name = "Unknown";
if ($input =~ /"name"\s*:\s*"([^"]+)"/) {
    $name = $1;
}

print "{\n";
print qq(  "status": "success",\n);
print qq(  "lang": "Perl",\n);
print qq(  "greeting": "Hello $name from Perl!",\n);
print qq(  "received_data": $input\n);
print "}\n";

File: src/Samvaad/services/polyglot/hello.cpp
#include <iostream>
#include <string>

using namespace std;

int main(int argc, char* argv[]) {
    if (argc < 2) {
        cout << "{\"error\": \"No arguments provided\"}" << endl;
        return 0;
    }

    string input = argv[1];
    string name = "Unknown";

    // Very basic JSON extraction for showcase
    size_t namePos = input.find("\"name\"");
    if (namePos != string::npos) {
        size_t colon = input.find(":", namePos);
        size_t quote1 = input.find("\"", colon);
        size_t quote2 = input.find("\"", quote1 + 1);
        if (quote1 != string::npos && quote2 != string::npos) {
            name = input.substr(quote1 + 1, quote2 - quote1 - 1);
        }
    }

    cout << "{" << endl;
    cout << "  \"status\": \"success\"," << endl;
    cout << "  \"lang\": \"C++\"," << endl;
    cout << "  \"greeting\": \"Hello " << name << " from C++!\"," << endl;
    cout << "  \"received_data\": " << input << endl;
    cout << "}" << endl;

    return 0;
}
